src/http/nodegen/middleware/inferResponseType.ts

```diff
@@ -0,0 +1,83 @@
+import express from 'express';
+import objectReduceByMap from 'object-reduce-by-map';
+import NodegenRequest from '@/http/interfaces/NodegenRequest';
+import getPreferredResponseFormat from '@/http/nodegen/utils/getPreferredResponseFormat';
+import GenerateItExpressResponse from '@/http/nodegen/interfaces/GenerateItExpressResponse';
+
+export default () => {
+
+  return (req: NodegenRequest, res: GenerateItExpressResponse, next: express.NextFunction) => {
+
+    res.inferResponseType = (
+      dataOrPath = undefined,
+      status = 200,
+      permittedTypes,
+      outputMap?
+    ) => {
+      // Send only a status when data is undefined
+      if (dataOrPath === undefined) {
+        return res.sendStatus(status);
+      }
+
+      permittedTypes = Array.isArray(permittedTypes) ? permittedTypes : [permittedTypes];
+
+      // Calculate the calculatedAcceptHeader based on the provided accept header
+      // When the accept header is not provided then we fallback to the 1st permittedType
+      // The last fallback is then application/json
+      const calculatedAcceptHeader = req.headers['accept'] ?
```

Comment:
`calculatedContentType` would be more accurate - and makes more below where `return res.set('Content-Type', calculatedAcceptHeader)`